pkg/template/servicebroker/provision.go

```diff
@@ -132,43 +167,61 @@ func (b *Broker) ensureBrokerTemplateInstance(namespace, instanceID string, didW
 }
 
 func (b *Broker) Provision(instanceID string, preq *api.ProvisionRequest) *api.Response {
+	if err := b.waitForReady(); err != nil {
+		return api.InternalServerError(err)
+	}
+
 	if errs := ValidateProvisionRequest(preq); len(errs) > 0 {
 		return api.BadRequest(errs.ToAggregate())
 	}
 
 	namespace := preq.Parameters[templateapi.NamespaceParameterKey]
 	impersonate := preq.Parameters[templateapi.RequesterUsernameParameterKey]
-
-	impersonatedKC, _, impersonatedTemplateclient, err := b.getClientsForUsername(impersonate)
-	if err != nil {
-		return api.InternalServerError(err)
-	}
+	u := &user.DefaultInfo{Name: impersonate}
 
 	template, err := b.lister.GetByUID(preq.ServiceID)
 	if err != nil {
 		return api.BadRequest(err)
 	}
 	if template == nil {
+		templates, err := b.lister.List(labels.Everything())
+		if err != nil {
+			return api.InternalServerError(err)
+		}
+		for _, t := range templates {
+			if string(t.UID) == preq.ServiceID {
+				template = t
+				break
+			}
+		}
+	}
+	if template == nil {
 		return api.BadRequest(kerrors.NewNotFound(templateapi.Resource("templates"), preq.ServiceID))
 	}
 	if _, ok := b.templateNamespaces[template.Namespace]; !ok {
 		return api.BadRequest(kerrors.NewNotFound(templateapi.Resource("templates"), preq.ServiceID))
 	}
 
-	lsar := authorizationapi.AddUserToLSAR(&user.DefaultInfo{Name: impersonate},
-		&authorizationapi.LocalSubjectAccessReview{
-			Action: authorizationapi.Action{
-				Verb:     "create",
-				Group:    templateapi.GroupName,
-				Resource: "templateinstances",
-			},
-		})
-	lsarResp, err := b.localSAR.LocalSubjectAccessReviews(namespace).Create(lsar)
-	if err != nil || lsarResp == nil || !lsarResp.Allowed {
-		if err == nil {
-			err = errors.New("forbidden")
+	// TODO: enable SAR for template - at the moment I think this doesn't work
+	// properly because group information isn't populated in u.
```

Comment:
does this mean that your information flow is incomplete somewhere?  That is a severe problem.